zanata-war/src/main/java/org/zanata/action/RoleSearch.java

```diff
@@ -4,20 +4,23 @@
 import java.util.List;
 
 import javax.annotation.PostConstruct;
+import javax.enterprise.context.SessionScoped;
 import javax.inject.Inject;
 import org.apache.deltaspike.core.api.exclude.Exclude;
 import org.apache.deltaspike.core.api.projectstage.ProjectStage;
 import javax.inject.Named;
 import org.zanata.seam.security.IdentityManager;
 import org.zanata.security.ZanataIdentity;
+import org.zanata.util.Synchronized;
 
 /**
  * @author Patrick Huang
  *         <a href="mailto:pahuang@redhat.com">pahuang@redhat.com</a>
  */
 @Named("zanataRoleSearch")
 // TODO this should probably be ViewScoped or even RequestScoped (plus xhtml changes)
-@javax.enterprise.context.SessionScoped
```

Comment:
Yes, someone still need to look at this and decide whether SessionScoped is the best option.